src/content/en/2019/markup.md

```diff
@@ -37,7 +37,7 @@ Instead, as in Hixie's original study,  what we will look at is how many sites i
 
 In 2005, Hixie's survey listed the top few most commonly used elements on pages.  The top 3 were `html`, `head` and `body` which he noted as interesting because they are optional and created by the parser if omitted.  Given that we use the post-parsed DOM,  they'll show up universally in our data.  Thus, we'll begin with the 4th most used element. Below is a comparison of the data from then to now (I've included the frequency comparison here as well just for fun).
 
-<figure id="fig1" data-markdown="1">
+<figure id="fig-1" data-markdown="1">
```

Comment:
Why are the fig IDs changing? I'm concerned about breaking existing links.

Reply:
For two of the chapters the fig ids were written in the markdown. For the rest they were generated in the script/. Unfortunately the ones which were handwritten were of a slightly different format and I need them to be the same format as really don't want to do a multi-line regex.

Will add the old links on the img itself to preserve old links. Bit messy but only two chapters.

Reply:
OK changed it to handle this:

```javascript
  //Some of our 2019 chapters (markdown and performance) had different fig ids
  const figcaption_regex = /<figcaption(.*?)>(.*?)([0-9]+)\./gi;
  if (html.includes('<figure id="fig1"')) {
    return html.replace(figcaption_regex, "<figcaption$1><a href=\"#fig$3\" class=\"anchor-link\">$2 $3.</a>");
  } else {
    return html.replace(figcaption_regex, "<figcaption$1><a href=\"#fig-$3\" class=\"anchor-link\">$2 $3.</a>");
  }
```